[PUPIL-1226] Display related subjects based on overrides

BTitterington opened the pull request:
## Description

Music year: {owa_music_year}

- List of changes

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-3316--oak-web-application.netlify.thenational.academy
2. Click on \_\_\_\_\_\_
3. You should see \_\_\_\_\_\_

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

RBrownsett commented:
@BTitterington looking good - just noticed that the text in the pupil area differs from the design (it has the same text as the teacher area in the PR)?
PR:
![Screenshot 2025-03-26 at 09 22 46](https://github.com/user-attachments/assets/22b70386-ef90-4678-b133-1a324a4b7379)

Design
![Screenshot 2025-03-26 at 09 23 01](https://github.com/user-attachments/assets/b0526fb2-86c0-41a0-a08e-82743efae105)

BTitterington commented:
@RBrownsett I might have got my wires crossed, and updated the pupil one with the copy I read on this ticket - https://www.notion.so/oaknationalacademy/OakLinkCard-copy-update-teacher-experience-1b426cc4e1b180438b57c9e01d154a6d?pvs=4

RBrownsett commented:
> @RBrownsett I might have got my wires crossed, and updated the pupil one with the copy I read on this ticket - https://www.notion.so/oaknationalacademy/OakLinkCard-copy-update-teacher-experience-1b426cc4e1b180438b57c9e01d154a6d?pvs=4

No worries - is it ok to update to match the designs?

weronika-szalas commented on src/components/TeacherComponents/RelatedSubjectsBanner/banners/FinancialEducationBanner.tsx:
```diff
@@ -0,0 +1,54 @@
+import { resolveOakHref } from "@/common-lib/urls";
+
+function FinancialEducationBanner({ keyStageSlug, isDesktop }: BannerProps) {
+  const programmeSlug = `financial-education-secondary-${keyStageSlug}`;
```
will this be always 'secondary', is there no way to pass programmeSlug as prop?

BTitterington replied:
I think you are right here. There either needs to be the phase passed through, or a conditional on the keystage value that updates that part of the slug. I'd say phase seems like the better approach.

BTitterington replied:
I've adjusted to use the phase in addition now. I couldn't use the programmeSlug as in Maths for instance it looks like this "maths-secondary-ks4-foundation" and the "foundation" part wouldn't be applicable to the finance slugs.

BTitterington commented:
> > @RBrownsett I might have got my wires crossed, and updated the pupil one with the copy I read on this ticket - https://www.notion.so/oaknationalacademy/OakLinkCard-copy-update-teacher-experience-1b426cc4e1b180438b57c9e01d154a6d?pvs=4
> 
> No worries - is it ok to update to match the designs?

@RBrownsett - I've updated the text based on the designs, hopefully that is the right copy now.
